src/main/java/com/wanikani/api/v2/request/SubjectsRequest.java

```diff
@@ -1,76 +1,28 @@
 package com.wanikani.api.v2.request;
 
-import com.wanikani.api.v2.model.SubjectType;
+import lombok.Singular;
+import lombok.experimental.SuperBuilder;
 
 import java.util.ArrayList;
 import java.util.List;
-import java.util.stream.IntStream;
-
-import static com.wanikani.api.v2.request.QueryStringUtils.append;
-import static com.wanikani.api.v2.request.QueryStringUtils.appendList;
-import static java.util.stream.Collectors.toList;
-
-public class SubjectsRequest implements Request {
-
-    private final String queryString;
-
-    public SubjectsRequest(String queryString) {
-        this.queryString = queryString;
-    }
-
-    public static Builder builder() {
-        return new Builder();
-    }
 
+@SuperBuilder(toBuilder = true)
+public class SubjectsRequest extends CollectionRequest {
+    @Singular
+    private final List<Integer> levels = new ArrayList<>();
+    @Singular
+    private final List<String> types = new ArrayList<>();
+    @Singular
+    private final List<String> slugs = new ArrayList<>();
+    private final Boolean hidden;
+
+    @Override
     public String getQueryString() {
-        return queryString;
-    }
-
-    public static class Builder extends CollectionBuilder<SubjectsRequest, Builder> {
-        private final List<Integer> levels = new ArrayList<>();
-        private List<String> types = new ArrayList<>();
-        private List<String> slugs = new ArrayList<>();
-        private Boolean hidden;
-
-        public Builder slugs(List<String> slugs) {
-            this.slugs = slugs;
-            return this;
-        }
-
-        public Builder hidden(Boolean hidden) {
-            this.hidden = hidden;
-            return this;
-        }
-
-        public Builder types(List<SubjectType> types) {
-            this.types = types.stream().map(SubjectType::getName).collect(toList());
-            return this;
-        }
```

Comment:
In the previous way, the user could pass in a list of `SubjectType` enums, but now it would require the user to map them to the SubjectType names manually. `SubjectMaterialsRequest#subjectTypes`, `ReviewStatisticsRequest#subjectTypes` and `AssignmentRequest#subjectTypes` also has this issue.

Is there a way for us to handle this case to allow the user to pass the enums?

Reply:
Good spot. This was just me being stupid and copy-pasting the old builder's fields. Fixed it by adding an overload where you pass a string mapping function

```
.appendList("subject_types", subjectTypes, SubjectType::getName)
```